docs/cli/ilab-local-config.md

```diff
@@ -0,0 +1,184 @@
+
+When we evaluate a particular model with `ilab eval <model-ref>`, we would then save the corresponding score in this new directory and link it back to the model that was evaluated and the metric.
+
+By linking a given model with an evaluation score, we could provide an `ilab eval list` command that can output scores
+for particular models.
+
+Here's a sample output of the `ilab eval list` command:
+
+```
+MODEL REF   BASE MODEL                    EVAL SCORE  METRIC     EVAL DATE 
+abcde      	ibm-granite/granite-7b-base   84.2%       MMLU       2024-06-24
+cbdea      	abcde                         78.9%       MMLU       2024-06-23
+fghij       ibm-granite/granite-7b-base   7.89        MT-Bench   2024-06-22
+```
+
+Since this would be JSON data, we don't expect it to be very expensive or consume a lot of space, so clearing it should not be a priority. 
+
+
+
+We could provide an `ilab eval clear` function that erases all of the evaluation data. We could also generate refs for the eval runs as well, and clear them using `ilab eval clear <eval-ref>`.
+
+The output would look something like
+
+
+## Evolving configs
+
+
+Since the instructlab CLI is an evolving community project, we expect the schemas and methods we use to change over time To maintain compatibility, we should have schema versions for our configs.
+
+For instance, our current one would be a `v1` and if we decide
+to add something in the future we would update it to a `v2`.
+
+
+In the future, we will also want to have config migration as we evolve the library to avoid breaking things for existing users.
+
+
+
+## Scope of this proposal
+
+
+This proposal provides a long-term goal of numerous components and how we expect them to interact with each other.
+
+The main point here is to move the config to live in a single place so that the `ilab` command can be executed from any directory with confidence. In addition, we require the use of refs and CRUD-style commands to simplify the management and interaction of data generated by `ilab`.
+
+The steps outlined to implement this are:
+
+1. Implement logic for the `~/.config/instructlab` and `~/.local/share/instructlab` directories, and expect the config to live and be read from there unless another config is manually specified. For simplicity, if the config we read in is located at `~/.config/instructlab` - the default location - then we'll assume relative paths should be resolved with respect to the data directory `~/.local/share/instructlab`.
+1. Implement logic to the existing commands so that datasets, models, and eval scores can be read from `ilab list`
+1. Implement ref logic and update all existing commands to be able to resolve objects from refs
+
+
+Based on these steps, #1 would be the trickiest because we assume that the paths referenced in the config file are relative to the calling process. 
+
+One approach here could be removing these paths from the default and forcing the user to specify them relatively.
```

Comment:
I am having trouble understanding what you mean here

Reply:
@n1hility My wording here is bad but the idea is simple:

- When the config read is the default one `~/.config/ilab/config.yaml` then parse paths relative to `~/.local/share/ilab`.
- When that's NOT the config, we would parse relative paths with respect to the calling process. E.g. if `ilab` is invoked in `/foo` and the `config.model_path` references a file in `bar/granite_7b.gguf` then it would resolve to `/foo/bar/granite_7b.gguf`.
- All absolute paths will be parsed as-is